diff -Naur ./include/gazeapi.h ../original/include/gazeapi.h
--- ./include/gazeapi.h	2014-04-24 15:42:50.000000000 +0100
+++ ../original/include/gazeapi.h	2014-07-11 17:54:27.000000000 +0100
@@ -101,7 +101,7 @@
          * \param[in] port port number to connect to server on.
          * \return bool True if connected, false if connection failed.
          */
-        bool connect(bool push_mode, unsigned short port);
+        bool connect(bool push_mode, std::string server, unsigned short port);
 
         /** Disconnect from server. */
         void disconnect();
diff -Naur ./src/gazeapi.cpp ../original/src/gazeapi.cpp
--- ./src/gazeapi.cpp	2014-04-24 15:42:50.000000000 +0100
+++ ../original/src/gazeapi.cpp	2014-07-11 17:55:34.000000000 +0100
@@ -178,7 +178,7 @@
         }
 
         // GazeApi support
-        bool connect(bool push_mode, std::string const & port)
+        bool connect(bool push_mode, std::string const & server, std::string const & port)
         {
             if (AS_STOPPED != m_state)
             {
@@ -187,7 +187,7 @@
 
             m_port = port;
 
-            bool const success = m_socket.connect("127.0.0.1", m_port);
+            bool const success = m_socket.connect(server.c_str(), m_port);
 
             if (success)
             {
@@ -745,14 +745,14 @@
 
     bool GazeApi::connect(bool push_mode)
     {
-        return connect(push_mode, 6555);
+        return connect(push_mode, "localhost", 6555);
     }
 
-    bool GazeApi::connect(bool push_mode, unsigned short port)
+    bool GazeApi::connect(bool push_mode, std::string server, unsigned short port)
     {
         std::stringstream ss;
         ss << port;
-        return m_engine->connect(push_mode, ss.str());
+        return m_engine->connect(push_mode, server, ss.str());
     }
 
     void GazeApi::disconnect()
